lib/source/consul.js

```diff
@@ -0,0 +1,292 @@
+/* global Log, Config */
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+const consul = require('consul');
+
+const DEFAULT_AGENT_ADDRESS = '127.0.0.1';
+const DEFAULT_AGENT_PORT = 8500;
+
+class Consul extends EventEmitter {
+  /**
+   * Creates a new instance of a Consul souce plugin
+   *
+   * The following options are valid:
+   * - host (String, default 127.0.0.1): Consul agent's address
+   * - port (Number, default 8500) Consul agent's HTTP(S) port
+   * - secure (Boolean, default: true): enable HTTPS
+   *
+   * @param {Object} options  Options that can be set for the plugin
+   */
+  constructor(options) {
+    super();
+
+    const configuration = options || Object.create(null);
```

Comment:
I've been meaning to ask, because I've seen you use this pattern, but what's the impetus for using `Object.create(null)`?